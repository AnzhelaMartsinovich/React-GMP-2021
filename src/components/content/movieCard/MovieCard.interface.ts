export interface MovieCardProps {
  getMovieDataRequest: (id: number) => void;
}

export interface MovieItemProps {
  id: number;
  poster_path: string;
  title: string;
  genres: string[];
  release_date: string;
}
