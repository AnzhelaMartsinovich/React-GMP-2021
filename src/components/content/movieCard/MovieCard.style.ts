import { styled } from '../../../theme';

export const MovieCardContainer = styled.div`
  display: grid;
  grid-gap: 30px;
  grid-template-columns: repeat(auto-fill, 400px);
  justify-content: space-between;
  grid-auto-rows: 1fr;
`;
