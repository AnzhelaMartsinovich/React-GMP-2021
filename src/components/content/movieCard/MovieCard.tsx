import React, { FC } from 'react';

import { MovieCardProps } from './MovieCard.interface';
import { MovieCardItem } from './movieCardItem/MovieCardItem';

import { MovieCardContainer } from './MovieCard.style';

export const MovieCard: FC<MovieCardProps> = ({ moviesData }) => (
  <MovieCardContainer>
    {moviesData.map(({ id, poster_path, title, genres, release_date }) => (
      <MovieCardItem
        key={id}
        id={id}
        poster_path={poster_path}
        title={title}
        genres={genres}
        release_date={release_date}
      />
    ))}
  </MovieCardContainer>
);
