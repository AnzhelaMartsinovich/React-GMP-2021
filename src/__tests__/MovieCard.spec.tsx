import React from 'react';
import { render } from '@testing-library/react';

import { MovieCard } from '../components/content/movieCard/MovieCard';
import { MoviesDataProps } from 'baseTypes/BaseTypes.interface';

const moviesData: MoviesDataProps = {
  moviesData: [
    {
      id: 0,
      title: '',
      tagline: '',
      vote_average: 0,
      vote_count: 0,
      release_date: '',
      poster_path: '',
      overview: '',
      budget: 0,
      revenue: 0,
      genres: [''],
      runtime: 0,
    },
  ],
  getMovieDataRequest: jest.fn(),
};

describe('MovieCard tests', () => {
  it('MovieCard renders correctly', () => {
    const movieCard = render(
      <MovieCard
        moviesData={moviesData.moviesData}
        getMovieDataRequest={moviesData.getMovieDataRequest}
      />
    );
    expect(movieCard).toMatchSnapshot();
  });
});
