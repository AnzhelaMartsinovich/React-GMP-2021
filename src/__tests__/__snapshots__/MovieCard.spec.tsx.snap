// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard tests MovieCard renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: initial;"
  >
    <div>
      <div
        class="sc-jcFkyM kSKNcR"
      >
        <div
          class="sc-cCcYRi hEvaZv"
        >
          <div
            class="sc-cidCJl gOhbOM"
          >
            <div
              class="sc-ikJzcn eUaIuh"
            >
              <img
                alt="dots"
                class="sc-gWXaA-D hhwFjC"
                src="test-file-stub"
              />
            </div>
            <img
              alt="New movie title"
              class="sc-jRQAMF dRupYH"
              src="https://m.media-amazon.com/images/M/MV5BOGYxYzZkMWQtNjJkMy00NTlkLWExNWMtOTNhMTg4MDcxNmU3XkEyXkFqcGdeQXVyMDk5Mzc5MQ@@._V1_.jpg"
            />
          </div>
          <div
            class="sc-bdvvaa hBHiVW"
          >
            <div
              class="sc-gsDJrp"
            >
              <h3
                class="sc-hKwCoD"
              >
                New movie title
              </h3>
              <p
                class="sc-eCImvq kodcFI"
              >
                comedy, science
              </p>
            </div>
            <div
              class="sc-dkPtyc eCETnk"
            >
              2020.02.21
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jcFkyM kSKNcR"
    >
      <div
        class="sc-cCcYRi hEvaZv"
      >
        <div
          class="sc-cidCJl gOhbOM"
        >
          <div
            class="sc-ikJzcn eUaIuh"
          >
            <img
              alt="dots"
              class="sc-gWXaA-D hhwFjC"
              src="test-file-stub"
            />
          </div>
          <img
            alt="New movie title"
            class="sc-jRQAMF dRupYH"
            src="https://m.media-amazon.com/images/M/MV5BOGYxYzZkMWQtNjJkMy00NTlkLWExNWMtOTNhMTg4MDcxNmU3XkEyXkFqcGdeQXVyMDk5Mzc5MQ@@._V1_.jpg"
          />
        </div>
        <div
          class="sc-bdvvaa hBHiVW"
        >
          <div
            class="sc-gsDJrp"
          >
            <h3
              class="sc-hKwCoD"
            >
              New movie title
            </h3>
            <p
              class="sc-eCImvq kodcFI"
            >
              comedy, science
            </p>
          </div>
          <div
            class="sc-dkPtyc eCETnk"
          >
            2020.02.21
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
